File: src/main/java/LocCount/LocCount.java
package LocCount;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.util.List;
import java.util.stream.Stream;

public class LocCount {
    public static void run(String filename) {
        try {
            BufferedReader fileReader = new BufferedReader(new FileReader(filename));

            Stream<String> parsedLines = fileReader.lines();

            List<String> filteredLines = Filter.filter(parsedLines);

            int count = Counter.count(filteredLines);

            new Printer(System.out).print("Lines of code", count);
        } catch (FileNotFoundException e) {
            throw new IllegalArgumentException("File: " + filename + " does not exist.");
        }
    }
}

File: src/main/kotlin/loccount/LocCount.kt
package loccount

import printer.Printer


class LocCount(private val printer: Printer) {

    operator fun invoke(fileName: String) {
        val allLines = FileReader.readLines(fileName)
        val validLines = LineFilter.filterCommentsAndEmptyLines(allLines)
        val lineCount = LineCounter.count(validLines)
        printer.print("Line count: $lineCount")
    }
}